journal.py

```diff
@@ -72,6 +78,17 @@ def modify(cls, eid=None, title=None, text=None, session=None):
         session.add(instance)
         return instance
 
+    def markd_in(self, text):
+        html_text = markdown(self.text, output_format='html5')
+
+        def my_highlight(matchobj):
+            return highlight(matchobj.group(0), PythonLexer(), HtmlFormatter())
+
+        pattern = r'(?<=<code>)[\s\S]*(?=<\/code>)'
+        html_text = re.sub(pattern, my_highlight, html_text)
+
+        return html_text
```

Comment:
excellent choice to make this a method of the Entry class, by the way.  That's shown by how neatly it fits into your templates.